http-service/src/main/java/net/runelite/http/service/grandexchange/GrandExchangeService.java

```diff
@@ -0,0 +1,145 @@
+
+		try (Connection con = sql2o.open())
+		{
+			con.createQuery(CREATE_GRAND_EXCHANGE_PRICES).executeUpdate();
+		}
+	}
+
+	public GrandExchangeEntry insert(int itemId, GuidePriceResponse guidePrice)
+	{
+		Instant insertTime = Instant.now();
+
+		try (Connection con = sql2o.open())
+		{
+			con.createQuery("INSERT INTO ge_prices (item_id, overall, buying, buying_quantity, selling, "
+				+ "selling_quantity, last_update) values (:itemId, :overall, :buying, :buyingQ, :selling, "
+				+ ":sellingQ, :lastUpdate)")
+				.addParameter("itemId", itemId)
+				.addParameter("overall", guidePrice.getOverall())
+				.addParameter("buying", guidePrice.getBuying())
+				.addParameter("buyingQ", guidePrice.getBuyingQuantity())
+				.addParameter("selling", guidePrice.getSelling())
+				.addParameter("sellingQ", guidePrice.getSellingQuantity())
+				.addParameter("lastUpdate", Timestamp.from(insertTime))
+				.addColumnMapping("item_id", "itemId")
+				.addColumnMapping("buying_quantity", "buyingQuantity")
+				.addColumnMapping("selling_quantity", "sellingQuantity")
+				.addColumnMapping("last_update", "lastUpdate")
+				.executeUpdate();
+		}
+
+		return new GrandExchangeEntry(itemId, guidePrice, insertTime);
+	}
+
+	public GrandExchangeEntry update(int itemId, GuidePriceResponse guidePrice)
+	{
+		Instant updateTime = Instant.now();
+
+		try (Connection con = sql2o.open())
+		{
+			con.createQuery("UPDATE ge_prices SET overall = :overall, buying = :buying, buying_quantity = :buyingQ, "
+				+ "selling = :selling, selling_quantity = :sellingQ, last_update = :lastUpdate WHERE item_id = :itemId")
+				.addParameter("itemId", itemId)
+				.addParameter("overall", guidePrice.getOverall())
+				.addParameter("buying", guidePrice.getBuying())
+				.addParameter("buyingQ", guidePrice.getBuyingQuantity())
+				.addParameter("selling", guidePrice.getSelling())
+				.addParameter("sellingQ", guidePrice.getSellingQuantity())
+				.addParameter("lastUpdate", Timestamp.from(updateTime))
+				.addColumnMapping("item_id", "itemId")
+				.addColumnMapping("buying_quantity", "buyingQuantity")
+				.addColumnMapping("selling_quantity", "sellingQuantity")
+				.addColumnMapping("last_update", "lastUpdate")
+				.executeUpdate();
+		}
+
+		return new GrandExchangeEntry(itemId, guidePrice, updateTime);
+	}
+
+	public GrandExchangeEntry get(int itemId)
+	{
+		try (Connection con = sql2o.open())
+		{
+			return con.createQuery("SELECT item_id, overall, buying, buying_quantity, selling, selling_quantity, "
+				+ "last_update FROM ge_prices WHERE ge_prices.item_id = :itemId")
+				.addParameter("itemId", itemId)
+				.addColumnMapping("item_id", "itemId")
+				.addColumnMapping("buying_quantity", "buyingQuantity")
+				.addColumnMapping("selling_quantity", "sellingQuantity")
+				.addColumnMapping("last_update", "lastUpdate")
+				.executeAndFetchFirst(GrandExchangeEntry.class);
+		}
+	}
+
+	public GuidePriceResponse lookupItem(int itemId) throws IOException
```

Comment:
Doesn't OSB have a url to bulk download prices? This probably would not work too well due to the number of requests we get.